contrib/babelfishpg_common/sql/geography.sql

```diff
@@ -169,27 +177,100 @@ CREATE OR REPLACE FUNCTION sys.GEOGRAPHY(sys.bbf_varbinary)
 
 CREATE OR REPLACE FUNCTION sys.bbf_varbinary(sys.GEOGRAPHY)
 	RETURNS sys.bbf_varbinary
-	AS '$libdir/postgis-3','LWGEOM_to_bytea'
-	LANGUAGE 'c' IMMUTABLE STRICT PARALLEL SAFE;
+	AS $$
+	DECLARE
+        byte bytea;
+	BEGIN
+		byte := (SELECT CAST ($1 AS bytea));
+		RETURN (SELECT CAST (byte AS sys.bbf_varbinary)); 
+	END;
+	$$ LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE CAST (bytea AS sys.GEOGRAPHY) WITH FUNCTION sys.GEOGRAPHY(bytea) AS IMPLICIT;
-CREATE CAST (sys.GEOGRAPHY AS bytea) WITH FUNCTION sys.bytea(sys.GEOGRAPHY) AS IMPLICIT;
-CREATE CAST (sys.bbf_varbinary AS sys.GEOGRAPHY) WITH FUNCTION sys.GEOGRAPHY(sys.bbf_varbinary) AS IMPLICIT;
-CREATE CAST (sys.GEOGRAPHY AS sys.bbf_varbinary) WITH FUNCTION sys.bbf_varbinary(sys.GEOGRAPHY) AS IMPLICIT;
+CREATE OR REPLACE FUNCTION sys.GEOGRAPHY(sys.bbf_binary)
+	RETURNS sys.GEOGRAPHY
+	AS $$
+	DECLARE
+        varBin sys.bbf_varbinary;
+	BEGIN
+		varBin := (SELECT CAST (CAST ($1 AS sys.VARCHAR) AS sys.bbf_varbinary));
+		-- Call the underlying function after preprocessing
+		RETURN (SELECT CAST (varBin AS GEOGRAPHY)); 
+	END;
+	$$ LANGUAGE plpgsql IMMUTABLE STRICT PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION sys.GEOGRAPHY(sys.GEOMETRY)
+CREATE OR REPLACE FUNCTION sys.text(sys.GEOGRAPHY)
+	RETURNS text
+	AS $$
+	BEGIN
+		RAISE EXCEPTION 'Conversion from data type Geography to Text is not allowed.';
```

Comment:
Is there an existing SQL Server error message that we can map this more precisely to?

Reply:
Done. Specified the type of CAST (i.e. explicit or implicit) in this error message and this error message is already picked from SQL Server.